storage/reads/array_cursor.go

```diff
@@ -16,18 +17,24 @@ func (v *singleValue) Value(key string) (interface{}, bool) {
 	return v.v, true
 }
 
-func newAggregateArrayCursor(ctx context.Context, agg *datatypes.Aggregate, cursor cursors.Cursor) cursors.Cursor {
+func newAggregateArrayCursor(ctx context.Context, agg *datatypes.Aggregate, cursor cursors.Cursor) (cursors.Cursor, error) {
+	return newWindowAggregateArrayCursor(ctx, agg, execute.Window{}, cursor)
+}
+
+func newWindowAggregateArrayCursor(ctx context.Context, agg *datatypes.Aggregate, window execute.Window, cursor cursors.Cursor) (cursors.Cursor, error) {
 	if cursor == nil {
-		return nil
+		return nil, nil
 	}
-
 	switch agg.Type {
 	case datatypes.AggregateTypeSum:
-		return newSumArrayCursor(cursor)
+		return newSumArrayCursor(cursor), nil
 	case datatypes.AggregateTypeCount:
-		return newCountArrayCursor(cursor)
+		return newCountArrayCursor(cursor), nil
+	case datatypes.AggregateTypeMean:
+		return newWindowMeanArrayCursor(cursor, window)
 	default:
 		// TODO(sgc): should be validated higher up
+		// TODO(faith): add other aggs to this switch as we port them over
```

Comment:
Isn't this kind of implied by the `panic`? As we add more aggregates, we'll hit this panic, which would show us we need to add it?

Reply:
Yeah, that's true. I guess it was a mental note for me more than anything. I can remove it.